### PR TITLE
Add formatting options for vip logs

### PR DESCRIPTION
**Pull request by mjangda**
`json` and `csv` are supported. Default is "console" which just outputs timestamp and message

See CAFE-312

cc @chriszarate 

## Steps to Test

1. Check out PR.
1. Run `npm run build`

Invalid format:

```
./dist/bin/vip-logs.js @123 --format jso
Error:  Invalid format: jso. The supported formats are: csv, json, console.
Debug:  VIP-CLI v2.6.0, Node v12.18.4, darwin 20.6.0
```

No format:

```
./dist/bin/vip-logs.js @
2021-12-02T16:08:31.141418694Z This is a log message
```

JSON

```
./dist/bin/vip-logs.js @123 --format json
[
	{
		"timestamp": "2021-12-02T16:08:31.141472025Z",
		"message": "This is a log"
	}
]
```

CSV
```
./dist/bin/vip-logs.js @123 --format csv
"timestamp","message"
"2021-12-02T16:08:31.141472025Z","This is a log"
```

### CODE REVIEW

File: src/bin/vip-logs.js [hunk 1]
```diff
@@ -45,18 +41,46 @@ export async function getLogs( arg: string[], opt ): Promise<void> {
...
+
+	// Strip out __typename
+	logs = logs.map( log => {
+		const { timestamp, message } = log;
+
+		return { timestamp, message };
+	} );
 
-export function validateInputs( type: string, limit: number, env: Object ): void {
-	if ( ! env.isK8sResident ) {
```
Comment (mjangda) [inline src/bin/vip-logs.js:52]: Meant to put this in a separate PR 🤦

File: __tests__/bin/vip-logs.js [hunk 2]
```diff
@@ -65,9 +68,10 @@ describe( 'getLogs', () => {
 		expect( logsLib.getRecentLogs ).toHaveBeenCalledTimes( 1 );
 		expect( logsLib.getRecentLogs ).toHaveBeenCalledWith( 1, 3, 'app', 500 );
 
-		expect( console.log ).toHaveBeenCalledTimes( 2 );
-		expect( console.log ).toHaveBeenNthCalledWith( 1, '2021-11-05T20:18:36.234041811Z My container message 1' );
-		expect( console.log ).toHaveBeenNthCalledWith( 2, '2021-11-09T20:47:07.301221112Z My container message 2' );
+		expect( console.log ).toHaveBeenCalledTimes( 1 );
```
Comment (chriszarate) [inline __tests__/bin/vip-logs.js:69]: What's going on here? Why isn't it logging the first line anymore?

Comment (mjangda) [reply __tests__/bin/vip-logs.js:69]: We now build an array and then only output once.

Comment (chriszarate) [reply __tests__/bin/vip-logs.js:69]: Ah, sorry, missed the newline :)

File: __tests__/bin/vip-logs.js [hunk 3]
```diff
@@ -76,11 +80,87 @@ describe( 'getLogs', () => {
...
+		} );
+
+		expect( rollbar.error ).not.toHaveBeenCalled();
+	} );
+
+	it( 'should display the logs in the output with CSV format', async () => {
+		opts.format = 'csv';
+
+		logsLib.getRecentLogs.mockImplementation( async () => [
+			{ timestamp: '2021-11-05T20:18:36.234041811Z', message: 'My container message 1' },
```
Comment (chriszarate) [inline __tests__/bin/vip-logs.js:144]: It would be great to test this with some double quotes and newlines and such
```suggestion
			{ timestamp: '2021-11-05T20:18:36.234041811Z', message: 'My container message has "double quotes", \'single quotes\', commas, multiple\nlines\n, and	tabs' },
```

File: src/bin/vip-logs.js [hunk 4]
```diff
@@ -86,6 +109,7 @@ command( {
 } )
 	.option( 'type', 'The type of logs to be returned: "app" or "batch"', 'app' )
 	.option( 'limit', 'The maximum number of log lines', 500 )
+	.option( 'format', 'Output the log lines in CSV or JSON format', 'console' )
```
Comment (chriszarate) [inline src/bin/vip-logs.js]: Subjective, but I might suggest `plain` or `text` as the default format name. All formats get sent to the console. Naming the format `console` might imply that the other formats are written to disk.

File: src/bin/vip-logs.js [hunk 5]
```diff
@@ -99,5 +123,13 @@ command( {
 			usage: 'vip @mysite.production logs --limit 100',
 			description: 'Get the most recent 100 logs',
 		},
+		{
+			usage: 'vip @mysite.production logs --limit 100 --format csv',
+			description: 'Get the most recent 100 logs formatted as a CSV',
```
Comment (chriszarate) [inline src/bin/vip-logs.js]:
```suggestion
			description: 'Get the most recent 100 log entries formatted as comma-separated values (CSV)
```

File: src/bin/vip-logs.js [hunk 6]
```diff
@@ -99,5 +123,13 @@ command( {
...
+		},
+		{
+			usage: 'vip @mysite.production logs --limit 100 --format json',
+			description: 'Get the most recent 100 logs formatted in JSON',
```
Comment (chriszarate) [inline src/bin/vip-logs.js]:
```suggestion
			description: 'Get the most recent 100 log entries formatted as JSON',
```

File: src/bin/vip-logs.js [hunk 7]
```diff
@@ -99,5 +123,13 @@ command( {
 			usage: 'vip @mysite.production logs --limit 100',
 			description: 'Get the most recent 100 logs',
```
Comment (chriszarate) [inline src/bin/vip-logs.js]:
```suggestion
			description: 'Get the most recent 100 log entries',
```

File: src/bin/vip-logs.js (hunk 4 above)
Comment (mjangda) [reply src/bin/vip-logs.js]: `text` makes sense 👍

Comment (mjangda) [reply src/bin/vip-logs.js]: 8ec6b6b

File: __tests__/bin/vip-logs.js (hunk 3 above)
Comment (mjangda) [reply __tests__/bin/vip-logs.js:144]: Arguably, this should be left to the CSV formatting lib, but I guess it doesn't hurt to test here. I did change this to spy on the console output instead which is a better test.

89e802d

**Review by chriszarate — APPROVED** [review]
🎉

**Review by robersongomes — APPROVED** [review]
This looks great!
Thanks @mjangda and @chriszarate!
